### PR TITLE
Fix validation

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by jdef** [comment]
I just cherry-picked the first 3 commits. the 4th doesn't compile (missing log)

File: src/main/scala/mesosphere/marathon/core/volume/providers/DVDI.scala [hunk 1]
```diff
@@ -83,17 +83,22 @@ protected case object DVDIProvider extends InjectionHelper[PersistentVolume] wit
...
+          internalNameViolations.toSet[Violation] ++ instanceViolations.toSet ++ ruleViolations.toSet))
       }
       if (groupViolations.isEmpty) Success
       else Failure(groupViolations.toSet)
     }
   }
 
-  def driversInUse(ct: Container): Set[String] =
-    DVDIProvider.this.collect(ct).flatMap(_.persistent.options.get(optionDriver)).toSet
+  def driversInUse(ct: Container): Set[String] = {
```
Comment (sttts) [inline src/main/scala/mesosphere/marathon/core/volume/providers/DVDI.scala]: This part was not meant to be commited.

**Comment by aquamatthias** [comment]
@jdef looks like this PR can be closed then?

**Comment by sttts** [comment]
Yes.

**Pull request closed without merging**